source/vibe/core/net.d

```diff
@@ -789,9 +789,16 @@ struct TCPConnection {
 		size_t nbytes = 0;
 		m_context.readTimeout.loopWithTimeout!((remaining) {
 			if (m_context.readBuffer.length == 0) {
-				if (mode == IOMode.immediate || mode == IOMode.once && nbytes > 0)
+				if (mode == IOMode.immediate || (mode == IOMode.once && nbytes > 0))
 					return true;
-				enforce(waitForData(remaining), "Reached end of stream while reading data.");
+				auto ret = waitForDataEx(remaining);
+				if(ret  == WaitForDataStatus.timeout) {
+					// should throw ReadTimeoutException
+					return true;
```

Comment:
Shouldn't this be `throw new ReadTimeoutException("Read operation timed out")` instead?

Reply:
the "ReadTimeoutException" will throw by loopWithTimeout. it should "return false";
I will fix and add test.